pkg/reconciler/pipelinerun/resources/workspaces.go

```diff
@@ -0,0 +1,56 @@
+package resources
+
+import (
+	"fmt"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+)
+
+func ResolveWorkspaces(p *v1alpha1.PipelineSpec) (*v1alpha1.PipelineSpec, error) {
+	spec := p.DeepCopy()
```

Comment:
maybe a comment explaining why a deepcopy is needed? (im guessing b/c we're going to start making changes to it?)